airflow/api/client/local_client.py

```diff
@@ -51,3 +51,83 @@ def create_pool(self, name, slots, description):
     def delete_pool(self, name):
         p = pool.delete_pool(name=name)
         return p.pool, p.slots, p.description
+
+    def add_connection(self, conn_id,
+                       conn_uri=None,
+                       conn_type=None,
+                       conn_host=None,
+                       conn_login=None,
+                       conn_password=None,
+                       conn_schema=None,
+                       conn_port=None,
+                       conn_extra=None):
+        """
+
+        :param conn_id:
+        :param conn_uri:
+        :param conn_type:
+        :param conn_host:
+        :param conn_login:
+        :param conn_password:
+        :param conn_schema:
+        :param conn_port:
+        :param conn_extra:
+        :return: The new Connection
+        """
+        return connections.add_connection(conn_id,
```

Comment:
So the idea here was the client(s) would return the same objects (regardless of client type) and leave formatting the response to the api (cli or http). What about returning the json representation from the client, (which the local and json clients could do) and still leaving formatting to the API. TBH, the differences in clients confuses me and I'm still not sure what  or why we have two different clients given that the end users seem to interact with the API rather than the client directly. Any clarity you could provide there would be appreciated.